rs/messaging/src/routing/stream_builder/tests.rs

```diff
@@ -1057,13 +1057,9 @@ fn canister_states_with_outputs<M: Into<RequestOrResponse>>(
         match msg {
             RequestOrResponse::Request(req) => {
                 // Create a matching callback, so that enqueuing any reject response will succeed.
-                register_callback(
-                    canister_state,
-                    req.sender,
-                    req.receiver,
-                    req.sender_reply_callback,
-                    req.deadline,
-                );
+                let callback_id =
+                    register_callback(canister_state, req.sender, req.receiver, req.deadline);
+                assert_eq!(req.sender_reply_callback, callback_id);
```

Comment:
Turns out it duplicates the code used in the `CallContextManager` for generating [test messages](https://sourcegraph.com/github.com/dfinity/ic@rumenov/regtypes/-/blob/rs/messaging/src/routing/stream_builder/tests.rs?L1010). I'd rather do any deeper changes to the testing framework, at least no in this PR, but adding this assert will at least let us know if the two implementations diverge.

Reply:
Could we maybe have `msgs` be internally generated in this function as opposed to being provided as an input parameter?

Reply:
These messages are used for other things first in some places. I guess you could but with a lot more refactoring than I want to do here.

Reply:
Sounds good. Then lets document the assumption for why this assert holds in a comment just before the comment.

Reply:
Put a comment there explaining the consistency check.

Reply:
A bit late, but you could also make `msgs` mutable and set their callback IDs to the ones we generate here.

Or do the callers also have expectations regarding the exact callback IDs they provide?